pkg/tdex-market-loader/client.go

```diff
@@ -0,0 +1,322 @@
+	"github.com/tdex-network/tdex-protobuf/generated/go/trade"
+	"github.com/tdex-network/tdex-protobuf/generated/go/types"
+	"golang.org/x/net/proxy"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/insecure"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"strings"
+)
+
+const (
+	onionUrlRegex = "onion"
+	httpRegex     = "http://"
+	httpsRegex    = "https://"
+)
+
+type Service interface {
+	FetchProvidersMarkets(ctx context.Context) ([]LiquidityProvider, error)
+	FetchBalance(ctx context.Context, market Market) (*Balance, error)
+	FetchPrice(ctx context.Context, market Market) (*Price, error)
+}
+
+type tdexMarketLoaderService struct {
+	torProxyUrl string
+	registryUrl string
+}
+
+func NewService(torProxyUrl, registryUrl string) Service {
+	return &tdexMarketLoaderService{
+		torProxyUrl: torProxyUrl,
+		registryUrl: registryUrl,
+	}
+}
+
+func (t *tdexMarketLoaderService) FetchProvidersMarkets(
+	ctx context.Context,
+) ([]LiquidityProvider, error) {
+	res := make([]LiquidityProvider, 0)
+	liquidityProviders, err := t.fetchLiquidityProviders()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, v := range liquidityProviders {
+		markets, err := t.fetchLiquidityProviderMarkets(ctx, v)
+		if err != nil {
+			log.Errorf(
+				"error while trying to fetch markets for liquidity provider: %v, err: %v\n",
+				v.Name,
+				err.Error(),
+			)
+			continue
+		}
+		res = append(res, LiquidityProvider{
+			Name:     v.Name,
+			Endpoint: v.Endpoint,
+			Markets:  markets,
+		})
+	}
+
+	return res, nil
+}
+
+func (t *tdexMarketLoaderService) FetchBalance(
+	ctx context.Context,
+	market Market,
+) (*Balance, error) {
+	conn, close, err := t.getConn(market.Url)
+	if err != nil {
+		return nil, err
+	}
+	defer close()
+
+	client := trade.NewTradeClient(conn)
+	reply, err := client.MarketPrice(ctx, &trade.MarketPriceRequest{
+		Market: &types.Market{
+			BaseAsset:  market.BaseAsset,
+			QuoteAsset: market.QuoteAsset,
+		},
+		Type:   trade.TradeType_SELL,
+		Amount: 1000,
+		Asset:  market.BaseAsset,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &Balance{
+		BaseBalance:  int(reply.GetPrices()[0].GetBalance().GetBaseAmount()), //TODO to be update with new spec
+		QuoteBalance: int(reply.GetPrices()[0].GetBalance().GetQuoteAmount()),
+	}, nil
+}
+
+func (t *tdexMarketLoaderService) FetchPrice(
+	ctx context.Context,
+	market Market,
+) (*Price, error) {
+	conn, close, err := t.getConn(market.Url)
+	if err != nil {
+		return nil, err
+	}
+	defer close()
+
+	client := trade.NewTradeClient(conn)
+	reply, err := client.MarketPrice(ctx, &trade.MarketPriceRequest{
+		Market: &types.Market{
+			BaseAsset:  market.BaseAsset,
+			QuoteAsset: market.QuoteAsset,
+		},
+		Type:   trade.TradeType_SELL,
+		Amount: 1000,
+		Asset:  market.BaseAsset,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &Price{
+		BasePrice:  int(reply.GetPrices()[0].GetPrice().GetBasePrice()), //TODO to be update with new spec
```

Comment:
```suggestion
		 //TODO to be update with new spec
```

No, this will remain an array of prices, Balances will only change

Reply:
Done